src/requirements.h

```diff
@@ -190,6 +210,8 @@ struct requirement_data {
         const requirement_data disassembly_requirements() const;
 
     private:
+        std::string id_;
```

Comment:
> The id shouldn't care where it's stored, the string_id does not only provide access to the underlying type, but also provides a hint of what type of id it is.

How could you call `requirement_data::id()` and not know that you had a `requirement_data`?